source/nodejs/adaptivecards/src/carousel.ts

```diff
@@ -419,6 +452,27 @@ export class Carousel extends Container {
     }
 }
 
+export class CarouselEvent {
+    type : Enums.CarouselInteractionEvent;
+    carouselId : string | undefined;
+    activeCarouselPageId : string | undefined;
+    activeCarouselPageIndex : number | undefined;
+    constructor(carousel : Carousel) {
+        this.carouselId = carousel.id;
+        this.activeCarouselPageIndex = carousel.currentPageIndex;
+        this.activeCarouselPageId = (this.activeCarouselPageIndex != undefined) ? carousel.getItemAt(this.activeCarouselPageIndex).id : undefined; 
```

Comment:
I guess the correct way to check undefined is `typeof data !== "undefined"`

Reply:
pageIndex can be either a number or undefined. as long as it's not undefined, we can get correct carousel page.